src/components/containers/EditStudentContainer.js

```diff
@@ -0,0 +1,225 @@
+import { Component } from 'react';
+import { connect } from 'react-redux';
+import { Redirect } from 'react-router-dom';
+
+import { EditStudentView } from '../views';
+import {
+    addStudentThunk,
+    fetchStudentThunk,
+    editStudentThunk
+} from '../../store/thunks';
+
+class EditStudentContainer extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            firstname: "",
+            lastname: "",
+            imageUrl: "",
+            email: "",
+            gpa: "",
+            redirect: false,
+            redirectId: null,
+            errors: {
+                firstname: null,
+                lastname: null,
+                email: null,
+                gpa: null,
+                validate: null,
+                shouldDisplayError: false
+
+
+            }
+        };
+    }
+
+    /* Handle Focus for UX */
+    handleFocus = () => {
+        if (!this.state.errors.shouldDisplayError) {
+            this.setState(prevstate => ({
+                errors: {
+                    ...prevstate.errors,
+                    shouldDisplayError: true
+                }
+            }));
+        }
+    }
+
+    /* Helper Function */
+    isBlank(input) {
+        if (input.trim() === "")
+            return true
+        return false
+    }
+
+    /* Input Validation */
+    /*
+    * The first name, last name, and email fields are required.
+    */
+    handleChange = event => {
+        this.setState({
+            [event.target.name]: event.target.value.trim()
+        });
+
+        let firstnameError = false;
+        let lastnameError = false;
+        let emailError = false;
+
+        switch (event.target.name) {
+            case "firstname":
+                if (this.isBlank(event.target.value)) {
+                    firstnameError = true;
+                }
+                if (this.isBlank(this.state.email)) {
+                    emailError = true;
+                }
+                if (this.isBlank(event.target.value)) {
```

Comment:
The if statement should not contain event.target.value here, it should only contain that for the specific case

Reply:
not sure what you mean by this? i should change event.target.value to event.target.name?

Reply:
No -- take a look at the branch allstudents, it should have the correct case checking atm. Basically, for example, when you have case "firstname" -- that means that your event is changing the firstname input. So you want to use the input for the isBlank function, aka if (this.isBlank(event.target.value)). But for emailError and lastnameError under the same case, you don't use `event.target.value` because that only refers to the changed firstname. Use `this.state.email` and `this.state.lastname` instead.